### DISCUSSION TITLE
Meshopt compressed models break on iOS AR-Quicklook

### DISCUSSION THREAD

**Discussion by Drohnendon** [google/model-viewer#4455] [Q&A]
Hihi,

title says it all: My mesh opt-compressed glbs are rendered perfectly fine on web across a wide range of browsers and on android in AR. 
Once again only on iOS AR-Quicklook the models appear with lots of artifacts. 

Working model: 
![working](https://github.com/google/model-viewer/assets/49596695/03fe2962-c9c0-4a59-ba93-c90ec685b8f0)

same file on quicklook: 
![artifacts](https://github.com/google/model-viewer/assets/49596695/8581b61d-0f18-4b20-bea8-3f6edc238703)

Does anyone know a solution to this?

**Comment by elalish** [comment]
This looks like a three.js bug, probably in USDZExporter. Want to try filing it there? Make sure to include your model.

**Reply by Drohnendon** [reply]
All right, thank you Emmett! I am not very deep into three.js. What are keywords and phrases you think I should include in the bug report and / or my search for similar issues and discussions? Can you just go a little bit deeper into why you think this is a three.js bug and nothing about modelviewer?

**Reply by elalish** [reply]
Because we transparently use the three.js GLTFLoader and USDZExporter, and since it's rendering correctly in MV, the loader worked. See if you can reproduce it in their editor: https://threejs.org/editor/ - which can also import glTF and export USDZ with the same functions. I think `meshopt` and `USDZExporter` is probably all you need.
